### PR TITLE
Add an option to load the PowerShell profile

### PR DESCRIPTION
**Pull request by ofosos**
Hey Team,
I would like to be able to load the PowerShell profile on the agent. This change is a step in that direction.

Please be kind. I ran the unit tests, but I'm not sure this will work, any ideas on how to test this properly?

Cheers, Mark

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java [hunk 1]
```diff
@@ -48,6 +48,7 @@
 public final class PowershellScript extends FileMonitoringTask {
     private final String script;
     private String powershellBinary = "powershell";
+    private boolean loadProfile = false;
```
Comment (timja) [inline src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java:51]: false is the default
```suggestion
    private boolean loadProfile;
```

File: src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java [hunk 2]
```diff
@@ -63,6 +64,11 @@ public void setPowershellBinary(String powershellBinary) {
         this.powershellBinary = powershellBinary;
     }
 
+    public boolean isLoadProfile() { return loadProfile; }
```
Comment (timja) [inline src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java:67]: use line breaks, to match the code style in the file
```suggestion
    public boolean isLoadProfile() {
        return loadProfile; 
    }
```

File: src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java [hunk 3]
```diff
@@ -63,6 +64,11 @@ public void setPowershellBinary(String powershellBinary) {
...
+
+    @DataBoundSetter
+    public void setLoadProfile(boolean loadProfile) { this.loadProfile = loadProfile; }
```
Comment (timja) [inline src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java:70]: use line breaks, to match the code style in the file
```suggestion
    public void setLoadProfile(boolean loadProfile) {
        this.loadProfile = loadProfile;
    }
```

File: src/main/resources/org/jenkinsci/plugins/durabletask/PowershellScript/config.jelly [hunk 4]
```diff
@@ -7,4 +7,7 @@
     <f:entry field="powershellBinary" title="Executable">
         <f:select />
     </f:entry>
+    <f:entry field="loadProfile" title="Load Profile">
```
Comment (timja) [inline src/main/resources/org/jenkinsci/plugins/durabletask/PowershellScript/config.jelly:10]: you could also help add text by creating a file called:
help-loadProfile.html in the same directory

only if you think it's needed though

**Comment by ofosos** [comment]
Hey, @timja I incorporated your changes.

File: src/main/resources/org/jenkinsci/plugins/durabletask/PowershellScript/config.jelly (hunk 4 above)
Comment (ofosos) [reply src/main/resources/org/jenkinsci/plugins/durabletask/PowershellScript/config.jelly:10]: I think it's rather obvious, what this switch does. So I would go without a help text. But thanks for the pointer!

**Comment by car-roll** [comment]
Looks fine to me. I think you want to add some profile loading tests in https://github.com/jenkinsci/durable-task-plugin/blob/master/src/test/java/org/jenkinsci/plugins/durabletask/PowerShellCoreScriptTest.java

**Comment by ofosos** [comment]
Hey @car-roll, I added to test to see if the flag is present in the current process. That's not really nice, but it looks better than changing profiles on developer machines ;)

Is this sufficient for you?

**Comment by car-roll** [comment]
@ofosos admittedly I don't really know about PowerShell and  profiles. Is the test somehow validating that the profile has been successfully loaded? I just see assertions for script launch success.

**Comment by ofosos** [comment]
@car-roll The thing is, afaik we can't check that the profile has been loaded. Unless we modify the PowerShell profile and say include a variable. However it's kind of bad to modify every developers profile. So what we do instead is this: We spin up a powershell and the PowerShell script checks the flags of the current process (the powershell process) and returns success iff this includes the -NoProfile option. We check that the normal state is success and the return code is failure, when we pass the option to load the profiles.

**Comment by ofosos** [comment]
@car-roll ping

**Review by car-roll — APPROVED** [review]

**Comment by car-roll** [comment]
@timja looks like your comments were all addressed in this PR. Anything else you would like to add?

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
Lgtm

**Pull request merged**